tfjs-backend-webgpu/src/shader_util.ts

```diff
@@ -32,3 +32,24 @@ export function symbolicallyComputeStrides(
 
   return strides;
 }
+
+// atomicAdd only supports uint/int type. For float, we use
+// atomicCompareExchangeWeak to simulate.
+export const atomicAddSnippet =
+    (dst: string, index: string, val: string, component: number) => {
+      return `
+    for (var i = 0; i < ${component}; i = i + 1) {
```

Comment:
Is this really necessary to have a vectorized variant here? Can we achieve the same with `for (var i = 0; i < 4; i++) { atomicAdd(...); }` in `main()`?